logstash-core/src/main/java/org/logstash/ext/JRubyWrappedWriteClientExt.java

```diff
@@ -70,27 +69,44 @@ public JRubyWrappedWriteClientExt initialize(final ThreadContext context,
     }
 
     public JRubyWrappedWriteClientExt initialize(
-        final JRubyAbstractQueueWriteClientExt queueWriteClientExt, final String pipelineId,
-        final AbstractMetricExt metric, final IRubyObject pluginId) {
+        final JRubyAbstractQueueWriteClientExt queueWriteClientExt,
+        final String pipelineId,
+        final AbstractMetricExt metric,
+        final IRubyObject pluginId) {
         this.writeClient = queueWriteClientExt;
         // Synchronize on the metric since setting up new fields on it is not threadsafe
         synchronized (metric) {
             final AbstractNamespacedMetricExt eventsMetrics =
-                getMetric(metric, "stats", "events");
+                getMetric(metric,
+                        MetricKeys.STATS_KEY.asJavaString(),
+                        MetricKeys.EVENTS_KEY.asJavaString());
+
             eventsMetricsCounter = LongCounter.fromRubyBase(eventsMetrics, MetricKeys.IN_KEY);
-            eventsMetricsTime = LongCounter.fromRubyBase(eventsMetrics, PUSH_DURATION_KEY);
-            final AbstractNamespacedMetricExt pipelineMetrics =
-                getMetric(metric, "stats", "pipelines", pipelineId, "events");
-            pipelineMetricsCounter = LongCounter.fromRubyBase(pipelineMetrics, MetricKeys.IN_KEY);
-            pipelineMetricsTime = LongCounter.fromRubyBase(pipelineMetrics, PUSH_DURATION_KEY);
+            eventsMetricsTime = LongCounter.fromRubyBase(eventsMetrics, MetricKeys.PUSH_DURATION_KEY);
+
+            final AbstractNamespacedMetricExt pipelineEventMetrics =
+                getMetric(metric,
+                        MetricKeys.STATS_KEY.asJavaString(),
+                        MetricKeys.PIPELINES_KEY.asJavaString(),
+                        pipelineId,
+                        MetricKeys.EVENTS_KEY.asJavaString());
+
+            pipelineMetricsCounter = LongCounter.fromRubyBase(pipelineEventMetrics, MetricKeys.IN_KEY);
+            pipelineMetricsTime = LongCounter.fromRubyBase(pipelineEventMetrics, MetricKeys.PUSH_DURATION_KEY);
+
             final AbstractNamespacedMetricExt pluginMetrics = getMetric(
-                metric, "stats", "pipelines", pipelineId, "plugins", "inputs",
-                pluginId.asJavaString(), "events"
-            );
+                    metric,
+                    MetricKeys.STATS_KEY.asJavaString(),
+                    MetricKeys.PIPELINES_KEY.asJavaString(),
+                    pipelineId,
+                    MetricKeys.PLUGINS_KEY.asJavaString(),
+                    MetricKeys.INPUTS_KEY.asJavaString(),
+                    pluginId.asJavaString(), MetricKeys.EVENTS_KEY.asJavaString());
```

Comment:
:weary: `JRubyWrappedWriteClientExt::getMetric(AbstractMetricExt base,String...)` basically re-symbolizes each of the given `String`s, which is made much more apparent now that we are using all of the `MetricKeys`' `RubySymbol`s.

Would it be reasonable to provide and use overload?

~~~ java
    private static AbstractNamespacedMetricExt getMetric(final AbstractMetricExt base, final RubySymbol... keys) {
        return base.namespace(RubyUtil.RUBY.getCurrentContext(), RubyUtil.RUBY.newArray(keys));
    }
~~~